src/OrchardCore.Modules/OrchardCore.Tenants/Controllers/ApiController.cs

```diff
@@ -143,6 +144,51 @@ public async Task<IActionResult> Create(CreateApiViewModel model)
             return BadRequest(ModelState);
         }
 
+        [HttpPost]
+        [Route("edit")]
+        public async Task<IActionResult> Edit(TenantApiModel model)
+        {
+            if (!_currentShellSettings.IsDefaultShell())
+            {
+                return Forbid();
+            }
+
+            if (!await _authorizationService.AuthorizeAsync(User, Permissions.ManageTenants))
+            {
+                return this.ChallengeOrForbid("Api");
+            }
+
+            if (!_shellHost.TryGetSettings(model.Name, out var shellSettings))
+            {
+                return NotFound();
+            }
+
```

Comment:
@Piedone @hishamco 

What is missing here is the usage of `_tenantValidator.ValidateAsync()` that is used in the AdminController through `ValidateViewModelAsync(model, false)`.

Reply:
Right. Hisham, will you open a new PR?

Reply:
Fixed in #13160, also I did a little refactor to utilize the `ValidateModelAsync()` in both create and edit APIs